### PR TITLE
Allow choosing email app during signup and login

### LINKED ISSUE #15962: Allow the user to choose an email app during signup/login
### Expected behavior
When the users press the **Check email** button they should be able to choose an email app if more than one is available.

<details>
<summary>Expected behavior (PoC)</summary>

Some details on the PoC implementation [can be found here.](https://github.com/wordpress-mobile/WordPress-Android/pull/15911#issuecomment-1040431951)

https://user-images.githubusercontent.com/304044/154094725-2bada5d2-9e21-46f4-9200-a22b60440539.mp4
</details>

<details>
<summary>Note: this functionality is already available on iOS</summary>

IMG_0185
</details>

### Actual behavior
When the user presses the **Check email** button only the default email app opens without letting the user to choose an email app.

### Steps to reproduce the behavior
Note: You will need to have more than one email apps on your device (e.g. Gmail, Outlook)
1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide an email
3. On the next screen press the Check email button
4. *Notice* that you are not presented with an option to choose your email client

##### Tested on Pixel 5, Android 12, WPAndroid 19.2

### PR DESCRIPTION
**Pull request by antonis**
Fixes #15962

## Description
Adds functionality to allow the user to choose an email app (if more than one is available) when the *Check email* button is pressed during signup or login
 
### Signup: No email app available
Note: You will need to no email apps on your device (or have them disabled)
1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide an email
3. *Verify* that [there is no email button on the screen](https://github.com/wordpress-mobile/WordPress-Android/pull/15911)

<details>
<summary>Signup: No email apps available (Pixel 2 emulator/Android 9)</summary>

0apps
</details>

### Login: No email app available
Note: You will need to no email apps on your device (or have them disabled)
1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide an email of an existing user
3. Press the *Get a login link by email* button
4. On the next screen press the Check email button
5. *Verify* that the *Can't detect email client* error is presented

<details>
<summary>Login: No email apps available (Pixel 2 emulator/Android 9)</summary>

0login
</details>

### Signup/Login: One email app available
Note: You will need to have only one email app on your device
1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide an email
3. On the next screen press the Check email button
4. *Verify* that you progress directly to the only email app

<details>
<summary>One app available (Pixel 5/Android 12)</summary>

https://user-images.githubusercontent.com/304044/154718669-88152e09-e9a8-4b4a-bf68-d5f874fcc6c3.mp4
</details>

### Signup/Login: More email apps available
Note: You will need to have more than one email apps on your device (e.g. Gmail, Outlook)
1. Press the Signup button on the landing screen of the app (on a fresh install or logged out state)
2. Provide an email
3. On the next screen press the Check email button
4. *Verify* that you are presented with chooser to select one of the installed email apps.

#### Note: The email chooser UI can vary by device manufacturer and OS version. The chooser title is not displayed in all cases.

<details>
<summary>Two apps available (Pixel 5/Android 12/Multiple user profiles)</summary>

https://user-images.githubusercontent.com/304044/154718771-cacd3842-1bb6-4b53-bdb0-98efc3a759d1.mp4
</details>

<details>
<summary>Two apps available and title (Pixel 2 emulator/Android 9)</summary>

2appstitle
</details>

<details>
<summary>Four apps available (Huawei P9/Android 8)</summary>

4apps
</details>

## Regression Notes
1. Potential unintended areas of impact
Check email functionality during signup and login.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing since the `LoginActivity`/`LoginViewModel` has no test coverage for the enhanced functionality.

3. What automated tests I added (or what prevented me from doing so)
The added functionality depends on the the Android OS internals (e.g. querying available intents/apps) and cannot be unit tested easily.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt [hunk 1]
```diff
@@ -10,6 +10,7 @@
 * [**] Jetpack App: Signup using WordPress.com, create your favorite site and enjoy the Reader inside Jetpack app. [https://github.com/wordpress-mobile/WordPress-Android/pull/15946]
 * [*] Reader: Admins and Editors can edit comments from Reader [https://github.com/wordpress-mobile/WordPress-Android/pull/15957]
 * [*] Improves the error message shown when trying to create a new site with non-English characters in the domain name [https://github.com/wordpress-mobile/WordPress-Android/pull/15969]
+* [*] Allows choosing email app during signup and login [https://github.com/wordpress-mobile/WordPress-Android/pull/15979]
```
Comment (antonis) [inline RELEASE-NOTES.txt]: Note: Release note should be updated if this does not make it in `19.3`

**Review by ovitrif — APPROVED** [review]
I reviewed the code and tested using all 5 scenarios and it works great @antonis!

It's really nice that opening the e-mail app now works in the emulator (I tested on API 32). Previously this worked only on the physical device for me, while having Gmail installed in the emulator I would still get the "no e-mail apps" message.

One small issue I noticed which might be an actual use case in real life is related to hiding the **Check Email** button when there is no e-mail app on the device.
If I go to Play Store and install an e-mail client, when coming back (resuming the WP-Android app) to the signup &rarr; check email screen the button remains hidden.
The intent chooser screen works very nice with installing new e-mail clients and retrying the button.
The issue feels out of scope for this pull-request though, it might have been more useful to mention it on #15911.

**Comment by mkevins** [comment]
Hi @antonis :wave: :smile: 

Thank you for implementing this. I tested this out via the steps described (on a Pixel 3 API 30 - emulator), and it works as described - mostly. I found some strange behavior, depending on _which_ email apps I had installed. I have not completely investigated why this behavior differs, but I observed the following for when multiple apps were installed:

The three apps I installed (or had installed, Gmail came stock with the emulator) are: Gmail, [K-9 Mail](https://github.com/k9mail/k-9/releases/tag/5.911), and [ProtonMail](https://protonapps.com/protonmail-android). What I found was that for some reason, the chooser did not display K-9 mail, but only displayed Gmail and ProtonMail.

I also tried tapping a `mailto:` link in Chrome, and the same behavior was noted: only Gmail and ProtonMail were options in the chooser.

I then tried a different approach, to see if the results would be different, and I created a chooser from the original `ACTION_MAIN` intent we currently use for the query:

<details>
<summary>ACTION_MAIN approach</summary>

```patch
diff --git a/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java b/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
index 2f0162754f..ab4b74307f 100644
--- a/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
+++ b/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
@@ -149,7 +149,8 @@ public class WPActivityUtils {
     public static void openEmailClient(Context context) {
         Intent intent = Intent.makeMainSelectorActivity(Intent.ACTION_MAIN, Intent.CATEGORY_APP_EMAIL);
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
-        context.startActivity(intent);
+        Intent chooser = Intent.createChooser(intent, context.getString(R.string.login_select_email_client));
+        context.startActivity(chooser);
     }
 
     public static void disableReaderDeeplinks(Context context) {
diff --git a/WordPress/src/main/res/values/strings.xml b/WordPress/src/main/res/values/strings.xml
index 54810748ca..86126c98ec 100644
--- a/WordPress/src/main/res/values/strings.xml
+++ b/WordPress/src/main/res/values/strings.xml
@@ -3927,4 +3927,5 @@ translators: %s: Select control option value e.g: "Auto, 25%". -->
     <string name="about_automattic_logo_description">Automattic logo</string>
     <string name="about_automattic_back_icon_description">Back icon</string>
     <string name="about_automattic_app_icon_description">App icon</string>
+    <string name="login_select_email_client">Choose your email app:</string>
 </resources>
```
</details>

Interestingly, in this case, the chooser opens with only Gmail and K-9 mail apps, but _not_ ProtonMail. It isn't clear to me why this is the case, but I suspect it could be related with how various email apps advertise the intents which they can handle. It seems that there are some implicit trade-offs with either of these approaches :man_shrugging: 

This discovery makes me first wonder: is this behavior reproducible for you, and if so, whether we should test / audit how this behaves with various other email apps out there in the wild. Also, I wonder whether it's possible to somehow get the best of both worlds - perhaps by somehow building two lists and deduping them :thinking: .

**Comment by mkevins** [comment]
Hi Ovi :wave: :smile: 

> It's really nice that opening the e-mail app now works in the emulator (I tested on API 32). Previously this worked only on the physical device for me, while having Gmail installed in the emulator I would still get the "no e-mail apps" message.

Were you observing this behavior on the latest `trunk`? If so, I'm curious what email app(s) was/were installed at the time. :thinking: I believe there were issues previously regarding new limitations for querying activities in API 30 and above, though we recently resolved an issue with that - so I'm interested to know if there are scenarios that aren't accounted for in the latest implementation.

**Comment by antonis** [comment]
Than you both for reviewing 🙇 
I restrain from merging this now and dig a bit deeper on how to improve this further based on your feedback.

> Also, I wonder whether it's possible to somehow get the best of both worlds - perhaps by somehow building two lists and deduping them 🤔

I think that's a good idea Matthew 👍  I'll investigate this further 🙏

**Comment by ovitrif** [comment]
> Were you observing this behavior on the latest `trunk`? If so, I'm curious what email app(s) was/were installed at the time. 🤔 I believe there were issues previously regarding new limitations for querying activities in API 30 and above, though we recently resolved an issue with that - so I'm interested to know if there are scenarios that aren't accounted for in the latest implementation.

Hi @mkevins! 👋 😄 

Thank you for asking this! The issue was indeed only occurring before the fix for querying activities in API 30 & above.
On the latest trunk it works as expected 👍

**Comment by antonis** [comment]
> is this behavior reproducible for you, and if so, whether we should test / audit how this behaves with various other email apps out there in the wild

I too confirmed this behavior. Depending on the way we query different apps are returned.
Screenshot 2022-02-21 at 2 59 21 PM

I [drafted something quickly](https://github.com/wordpress-mobile/WordPress-Android/pull/15979/commits/10e6cc68d87dc5082712973ac75f91d9d29428e4) as a PoC of merging more than one query results. I'll keep the PR as draft for now to test this further.

<details>
<summary>Merged list screenshot</summary>

device-2022-02-21-185422
</details>

**Comment by mkevins** [comment]
> I [drafted something quickly](https://github.com/wordpress-mobile/WordPress-Android/pull/15979/commits/10e6cc68d87dc5082712973ac75f91d9d29428e4) as a PoC of merging more than one query results. I'll keep the PR as draft for now to test this further.

I haven't gotten around to testing this in an "end to end" way, but the code looks good, imo. I like that precedence was given to the intents / activities found via `ACTION_MAIN` rather than `ACTION_SENDTO`, since that's more semantically accurate, and it will default to the main activity for applications that are in both lists.

I did a little bit more digging to investigate the `intent-filter`s of K-9 and ProtonMail. It seems that ProtonMail does not use the `APP_EMAIL` category, but K-9 _does_, for it's [`MessageList` activity](https://github.com/k9mail/k-9/blob/49ceab2b6f5edbaeb39be54ad9cd33e84bb5b3ec/app/k9mail/src/main/AndroidManifest.xml#L194). I'm not sure _why_ ProtonMail (or any other email app, for that matter) would _not_ have that `intent-filter` category declared at least _somewhere_, but this outside of our control. :man_shrugging: 

ProtonMail [_does_ declare](https://github.com/ProtonMail/proton-mail-android/blob/402b2b5944dbaf55bd8a5540f61ee48916100d8a/app/src/main/AndroidManifest.xml#L237) a `SENDTO` action within an `intent-filter` for the `ComposeMessageActivity` activity, so this explains why it appears via the `SENDTO` query. K-9 [_also_ declares an activity with this action](https://github.com/k9mail/k-9/blob/49ceab2b6f5edbaeb39be54ad9cd33e84bb5b3ec/app/k9mail/src/main/AndroidManifest.xml#L215) for its `MessageCompose` activity, however, this is where things get interesting. This particular activity is [initially _disabled_](https://github.com/k9mail/k-9/blob/49ceab2b6f5edbaeb39be54ad9cd33e84bb5b3ec/app/k9mail/src/main/AndroidManifest.xml#L212), and [is only enabled after the number of accounts is greater than zero](https://github.com/k9mail/k-9/blob/c5e398901439d706812c6df294edbbf8b6a1e85c/app/core/src/main/java/com/fsck/k9/Core.kt#L40-L52).

In my testing, I have been installing email apps, but I have not been connecting them to an actual account. For some apps, (e.g. Gmail), intents are still advertised, and all roads there lead to the Gmail signup / login. For other apps, activities may be disabled until an account is added, so these will not appear in query result lists. I'm now wondering whether the chooser items found via `SENDTO` will launch their respective `email composition` activities when a user account is already present, when we'd rather, ideally, that those chooser items would launch their respective `message list` activities :thinking: . If so, I wonder if there is a robust way of launching the main (likely message list) activity of the application found via `SENDTO` :thinking: .

**Review by ovitrif — APPROVED** [review]
Thank you for this improvement @antonis!

I checked with many e-mail clients:

1. **Gmail**
    1. When it's the only email app on the device → It opens directly, on the emails list, as expected ✅
    2. When there are many e-mail apps on the device → It's the first, I can choose it and it opens as expected on the emails list ✅
2. **K-9 Mai**l → I can choose it and it opens as expected on the emails list ✅. Small problem I encountered but it has nothing to do with the WP app: I don't see any Gmail emails in this app.
3. **Samsung Email** → I can choose it and it opens as expected on the emails list ✅ Since I wasn't logged in, It opened first time on the "Add Email" screen, and I could go nicely through the flow to link my Gmail account then I ended up on the emails list.
4. **Yandex Mail** → I can choose it and it opens as expected on the emails list ✅
5. **ProtonMail** → I can choose it and it opens as expected on the emails list ✅ (had to sign in first, did it and ended up on the e-mails list)
6. **Email by Amobear Studio** → I can choose it and it opens as expected on the emails list ✅ (Though if not signed it, it doesn't ask me to sign in. I had to reopen the app and login manually, then it works from the 'Check Email' button).
7. **Outlook** → I can choose it and it opens as expected on the emails list ✅
8. **Spark** → I can choose it and it opens as expected on the emails list ✅
9. **Yahoo Mail** → I can choose it and it opens as expected on the emails list ✅ (though connection with my Gmail account wasn't working)
10. **GMX Mail** → I can choose it and it opens as expected ✅ (wasn't logged in)
11. **Blue Mail** → I can choose it and it opens as expected on the emails list ✅
12. **MyMail** → I can choose it and it opens as expected on the emails list ✅
13. **Mail.ru** → I can choose it and it opens as expected on the emails list ✅

> I'm now wondering whether the chooser items found via SENDTO will launch their respective email composition activities when a user account is already present

Hey @mkevins 👋 Thank you for this remark!
During my my testing all 13 e-mail clients opened in the Email list or at the Sign In step if I wasn't logged in 👍
Most of them worked quite nice with Sign In, I was able to sign in and the app landed on the Inbox, which is what we are looking for.

**Comment by mkevins** [comment]
Thanks Ovi for testing all of these! Since the logged in behavior may differ (based on the dynamic enabled state of the activities as discovered in the case of K-9, for example), I'm interested whether any of these apps would launch their compose activity when already logged into them at the time of tapping the "Check email".

I set up a test account with ProtonMail for this purpose, and observed that it wasn't even showing up in the list, I had K-9, Gmail, and ProtonMail installed, but only Gmail and K-9 were showing up in the list.

First, I tried adding actions `VIEW` and `SENDTO` to the queries in `AndroidManifest.xml`, but seemed to have no effect.

Next, I tried disabling Gmail, and uninstalling K-9, and then the ProtonMail app opened directly from the "Check email" button. It opened the message list, and not the compose activity, which is the behavior we want. But I'm confused as to why the visibility of ProtonMail was affected by the presence of the other apps.

I tried reinstalling K-9, and re-enabling Gmail, and sure enough ProtonMail disappeared from the list again. :thinking: I debugged the code as well, and confirmed that we are indeed passing a list of the 3 found activities, and deduping is working as expected as well. Somehow the chooser is not displaying all of the extra intents. I believe the reason for this is that [the `EXTRA_INITIAL_INTENTS` are limited to 2](https://developer.android.com/reference/android/content/Intent#EXTRA_INITIAL_INTENTS), and the empty intent that the chooser is created from has no action, so only the initial intents will be displayed (i.e. we are limited to 2). :man_shrugging: 

I'm not sure how likely it is that this would be an issue in practice, since I'm not sure how many users have more than 2 email apps installed. Unfortunately, all it takes for this to become an issue is for the app to be present - it does not even need to have real accounts associated with it. This means that for some users who are stuck with a stock app that they do not use but can't uninstall, it is possible that they use two other apps (e.g. Gmail and ProtonMail) and one of these apps will not be displayed. In my case, during my tests, I had not set up K-9, but did set up ProtonMail with a real account. However, ProtonMail was the only one that didn't show up (until I removed at least one of the other apps).

One approach to consider is to use an action with the chooser intent, but when I tried this using `emailAppIntent` from the `makeMainSelectorActivity`, the chooser had two _blank_ entries :thinking: and tapping them seemed to crash the activity :disappointed: .

**Comment by ovitrif** [comment]
Thanks Matthew for further digging into this, I wonder why it didn't happen in my case 🤷 . I could see all 13 email apps on my Samsung Galaxy S8+ (Android 9) that I'm using for testing.

I probably also forgot to mention that I had already logged in to some of the e-mail apps before doing the testing, to have as many different scenarios as possible, and for the apps where I was already logged in, I always landed on the messages list.

Maybe also worth to mention is that all the apps I tested have over 100 million users, I kinda picked from Play Store the most popular ones.

**Comment by mkevins** [comment]
> I kinda picked from Play Store the most popular ones.

Good thinking :smile: 

I picked a few that were open source so that any weird issues we were observing could be tracked down. I wonder, in your testing, which version of Android you are using? From the docs, we're limited to 2 initial intents in a `Chooser` from Android 10 and onward, but I imagine you are on Android 10+? If so, I'm not sure why the inconsistency :man_shrugging:  May be worth investigating further to fully understand the behavior differences.

**Comment by ovitrif** [comment]
Hey Matthew, thank you for following up on my comment!

> we're limited to 2 initial intents in a `Chooser` from Android 10 and onward

Somehow I missed this part, I noticed only the one highlighted in red:

docs

That's definitely the reason, since I don't get the update above **Android 9** on the Samsung S8+.
I'll have to order a new testing device for Android :)

**Comment by antonis** [comment]
@ovitrif @mkevins Thank you both for extensively testing and investigating this 🙇‍♂️ 

> I believe the reason for this is that [the EXTRA_INITIAL_INTENTS are limited to 2](https://developer.android.com/reference/android/content/Intent#EXTRA_INITIAL_INTENTS), and the empty intent that the chooser is created from has no action, so only the initial intents will be displayed (i.e. we are limited to 2).

Good catch 👍  I was able to reproduce this on my Pixel 5 (Android 12) but I'm not sure if there is a bullet proof solution for this ([without implementing a custom chooser](https://issuetracker.google.com/issues/134367295?pli=1)).

I experimented with starting with an `CATEGORY_APP_EMAIL` intent chooser and adding what was missing from the `ACTION_SENDTO`. This improves the situation a bit on Android 10+ since I can now show 4 apps but I'm [still missing two apps from the total 6 I had installed](https://user-images.githubusercontent.com/304044/155513872-e266bd4f-e5f8-44f6-a476-3db98b4b3fd6.png).

<details>
<summary>Experiment screenshot</summary>

device-2022-02-24-120535
</details>

<details>
<summary>Experiment diff</summary>

```diff
@@ -149,12 +149,13 @@ public class WPActivityUtils {
             return;
         }
         List<Intent> appIntents = new ArrayList();
-        for (ResolveInfo resolveInfo : queryEmailApps(context)) {
+        for (ResolveInfo resolveInfo : querySendEmailApps(context)) {
             Intent intent = context.getPackageManager().getLaunchIntentForPackage(resolveInfo.activityInfo.packageName);
             appIntents.add(intent);
         }
+        Intent emailAppIntent = Intent.makeMainSelectorActivity(Intent.ACTION_MAIN, Intent.CATEGORY_APP_EMAIL);
         Intent[] appIntentsArray = appIntents.toArray(new Intent[appIntents.size()]);
-        Intent chooserIntent = Intent.createChooser(new Intent(), title);
+        Intent chooserIntent = Intent.createChooser(emailAppIntent, title);
         chooserIntent.putExtra(Intent.EXTRA_INITIAL_INTENTS, appIntentsArray);
         context.startActivity(chooserIntent);
     }
@@ -196,6 +197,42 @@ public class WPActivityUtils {
         }
     }
 
+    private static List<ResolveInfo> querySendEmailApps(@NonNull Context context) {
+        PackageManager packageManager = context.getPackageManager();
+        List<ResolveInfo> intentsInfoList = new ArrayList();
+
+        // Get all apps with category email
+        Intent emailAppIntent = Intent.makeMainSelectorActivity(Intent.ACTION_MAIN, Intent.CATEGORY_APP_EMAIL);
+        List<ResolveInfo> emailAppIntentInfo =
+                packageManager.queryIntentActivities(emailAppIntent, PackageManager.MATCH_ALL);
+
+        // Get all apps that are able to send emails
+        Intent sendEmailAppIntent = new Intent(Intent.ACTION_SENDTO);
+        sendEmailAppIntent.setData(Uri.parse("mailto:"));
+        List<ResolveInfo> sendEmailAppIntentInfo =
+                packageManager.queryIntentActivities(sendEmailAppIntent, PackageManager.MATCH_ALL);
+
+        // Merge the two app lists
+        addNewIntents(intentsInfoList, emailAppIntentInfo, sendEmailAppIntentInfo);
+        return intentsInfoList;
+    }
+
+    private static void addNewIntents(List<ResolveInfo> list, List<ResolveInfo> existing, List<ResolveInfo> intents) {
+        for (ResolveInfo intent : intents) {
+            boolean found = false;
+            for (ResolveInfo item : existing) {
+                if (intent.activityInfo.applicationInfo.processName
+                        .equals(item.activityInfo.applicationInfo.processName)) {
+                    found = true;
+                    break;
+                }
+            }
+            if (!found) {
+                list.add(intent);
+            }
+        }
+    }
+
```
</details>

**Comment by ovitrif** [comment]
> I experimented with starting with an CATEGORY_APP_EMAIL intent chooser and adding what was missing from the ACTION_SENDTO. This improves the situation a bit on Android 10+ since I can now show 4 apps but I'm still missing two apps from the total 6 I had installed

That's an improvement to me! Thank you for experimenting to improve this solution 🙇 

@antonis  Should we go forward with this? The code doesn't look bad 🙂

**Comment by antonis** [comment]
Thank you for the feedback @ovitrif 🙇 

> Should we go forward with this? The code doesn't look bad 🙂

I'll clean up the code and prepare the PR for review :)

**Comment by antonis** [comment]
👋  @mkevins, @ovitrif thank you both for the feedback 🙇 
I worked around the [Android 10+ limitations](https://developer.android.com/reference/android/content/Intent#EXTRA_INITIAL_INTENTS) with 046ee00824f0ed5f2fb72a4488df72c258272837. [As suggested above](https://github.com/wordpress-mobile/WordPress-Android/pull/15979#issuecomment-1049766084) I start with a `CATEGORY_APP_EMAIL` intent chooser and add what was missing from the ACTION_SENDTO. This improves the situation but there is still a limitation on Android 10+ that can show app to 4 apps in my tests (was 2).

**Review by ovitrif — COMMENTED** [review]
The code looks good to me, good work @antonis!

Thank you for going the extra mile and re-iterate on the solution improve it and  🙇

I tested again with my entire collection of (13 😱) email apps on Samsung Galaxy S8+ (Android 9) and I noticed two new issues:

1. Two e-mail apps are duplicated in the list
   The two apps are both related to Mail.ru:

    <details>
    <summary>Duplicate entries in `intents` which end up in the `list`</summary>
    
    duplicatedIntents
    </details>
    
    <details>
    <summary>Preview Issue 1</summary>
    
    
    </details>
2. The app opened via the "Check email" chooser appears in background when tapping the button again after already opening one app from it.

   **Repro Steps:**
   - Press "Check Email" & select an app from the chooser
   - Go back to the WordPress app
   - Press "Check Email" again
   - **Notice** that the e-mail app opened before appears in the background.
   I don't think this needs to be fixed since it's specific to testing, and nothing is actually broken. The user can still pick an e-mail app from the chooser and login via the magic link.
   <br>
   <details>
    <summary>Preview Issue 2</summary>
    
    
    </details>

Do you think we should find a fix for the **Issue 1**?

**Comment by mkevins** [comment]
Hi Ovi, just adding my "two cents" about the additional issues you found :smiley: 

Issue 1 seems interesting.. are those duplicates part of the list from a single query (the manually queries `SEND_TO`)? Or only when combined w/ the Chooser that starts with `CATEGORY_APP_EMAIL` intent?

If it's the former, I guess we can dedup the manually generated list? If the latter, I wonder if we'd need to create another manual list for `CATEGORY_APP_EMAIL` just for the purposes of deduping the `SEND_TO` list, and then still proceed using the Chooser wrapping the `APP_EMAIL` intent (and initial manually added `SEND_TO` intents) :thinking: 

Regarding issue 2, I suspect this is an OS behavior (i.e. I think the Chooser is "in charge" of that). In any case, I agree that it doesn't seem like something worth "fixing". :+1:

**Comment by antonis** [comment]
@mkevins, @ovitrif Thank you both for iterating on this and providing more feedback 🙏 

> 1. Two e-mail apps are duplicated in the list

I was able to replicate this issue with the _Mail.ru_ app. What makes this app special (I guess there are other apps that behave similarly) is that it advertises more than one activities that can handle the `ACTION_SENDTO`.
Screenshot 2022-03-02 at 2 45 45 PM

I fixed this with 1d78dbf586a36f873b4bbf41873a9f1cd7f00917 by adding only the 1st occurrence or each app. 

Screenshot 2022-03-02 at 2 46 03 PM

> 2. The app opened via the "Check email" chooser appears in background when tapping the button again after already opening one app from it.

I was able to reproduce this too. This is the result of [using the `FLAG_ACTIVITY_NEW_TASK`](https://developer.android.com/reference/android/content/Intent#FLAG_ACTIVITY_NEW_TASK) for [the choose `Intent`](https://github.com/wordpress-mobile/WordPress-Android/pull/15979/files#diff-c7bb00a4fdc2d5fe1b25f684d96550baff53d6ddd3d4514ab10d0c2f74a25758R160). Moving the flag to the Main Selector Activity with 0c8577ad093bc31690474fba1adf738df3e550c9 seems to do the trick 🎉

**Review by ovitrif — APPROVED** [review]
Thank you @antonis for fixing both issues 🙇

Both solutions are neat 👌, I really like the final state of the code in this PR, awesome work once again with the last iteration 👍 !

- The email app chooser opened smoothly in all explorative testing scenarios thanks to the fix done by moving the FLAG_ACTIVITY_NEW_TASK flag over to the Main Selector Activity. It also seems to be logical to set it there, although I only realised this after seeing your fix in the code.
This confirms that [issue 2](https://github.com/wordpress-mobile/WordPress-Android/pull/15979#pullrequestreview-896209715) is fixed ✅

- Issue 1 is also fixed nicely ✅
I can see 13 email apps in the chooser, which is the number of email apps I have installed 🎉 
Both the Mail.ru app and the [Universal Email App](https://play.google.com/store/apps/details?id=park.hotm.email.app) which is based on the same source code are opening on the email list and I can login successfully by going to the most recent email and clicking the login link.
✅ This confirms that your approach to filter out the second advertised activity does the right job, since we end up with the correct activity to be launched from the chooser.

I'm confident the pull-request is ready to be merged now 🙇

**Comment by ovitrif** [comment]
@mkevins Thank you for sharing your view on the two issues!

> Issue 1 seems interesting.. are those duplicates part of the list from a single query (the manually queries SEND_TO)? Or only when combined w/ the Chooser that starts with CATEGORY_APP_EMAIL intent?

I hope the comment from Antonis [here](https://github.com/wordpress-mobile/WordPress-Android/pull/15979#issuecomment-1056957038) answered your question 👍 

It was indeed the result of the single query, `ACTION_SENDTO`, the value we provide for the `intents` argument of the `addNewIntents` function, [which is `sendEmailAppIntentInfo`](https://github.com/wordpress-mobile/WordPress-Android/blob/0c8577ad093bc31690474fba1adf738df3e550c9/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java#L177-L182) 😉

**Pull request merged**